packages/module/src/ExpandableText/ExpandableText.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import { Button, Stack, StackItem } from '@patternfly/react-core';
+import sanitizeHtml from 'sanitize-html';
+import { createUseStyles } from 'react-jss'
+
+const useStyles = createUseStyles({
+  expandableText: {
+    "& button": {
+      padding: "0"
+    },
+    "& is-inline": { marginRight: [ "10px", "0.625rem" ] }
+  } });
+
+const dangerousHtml = (html: string) => ({ __html: sanitizeHtml(html) });
+
+export interface ExpandableTextCustomButtonProps {
+  /** Allows a user to add a custom expand button */
+  expand: React.ReactNode;
+  /** Allows a user to add a custom collapse button */
+  collapse: React.ReactNode;
+}
+
+export interface ExpandableTextProps {
+  /** Additional classes added to the ExpandableText */
+  className?: string;
+  /** Text to display in the ExpandableText component */
+  text?: string;
+  /** A set length for the ExpandableText component. */
+  length?: number;
+  /** Indicates if the text should be kept inline */
+  inline?: boolean;
+  /** If set to true the expand text will be hidden */
+  hideExpandText?: boolean;
+  /** If set to true the text will be expanded when moused over */
+  expandOnMouseOver?: boolean;
+  /** Allows users to create custom expandable buttons */
+  customButton?: ExpandableTextCustomButtonProps;
+}
+
+const ExpandableText: React.FunctionComponent<ExpandableTextProps> = ({
+  text = '',
+  length = 150,
+  hideExpandText = false,
+  expandOnMouseOver = false,
+  className,
+  inline,
+  customButton,
+  ...props
+}: ExpandableTextProps) => {
+  const classes = useStyles();
+  const expandableTextClasses = clsx(classes.expandableText, className, { [`isInline`]: inline });
+  const trimmedText = text.substring(0, length);
+  const textOverflow = text.length > length;
+  const [ showText, setShowText ] = useState(false);
+  const toggleText = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event && event.preventDefault();
+    setShowText(!showText);
+  };
+
+  const cloneButton = (button: React.ReactNode) => React.isValidElement(button) ? React.cloneElement(button as React.ReactElement, { onClick: toggleText }) : 
+    <Button variant='link' onClick={toggleText}>
+      {button}
+    </Button>;
+
+  const expandButton = 
```

Comment:
I am in favor of splitting the expand button into a separate component and providing it as a default prop to a new `expandButton` prop. Adding props API alongside a custom node element API is IMO a duplicate. If we split the expand button we give the consumer the ability to customize every aspect of the button.

They will have the ability to use a pre-defined variant, customize the props of the predefined variant by importing and changing its props, and use an entirely custom node if they choose to.

Reply:
I don't think we would need to make an expandButton component if we go this route.  They could use a PF Button and pass that to use.  Only thing the button did internally for the component is control showing the state, however if we are not going to control the state then I don't see the reason to create an expandButton component.

Reply:
> I don't think we would need to make an expandButton component if we go this route. 

That is a valid point. Part of this is also checking if we actually need components like that.  We have already identified several components that are "useless" and we will discontinue them from FEC. Maybe this component should be just a demo in some docs.

Reply:
I think so @fhlavac brought this up as well last week if we would need this component.  I think having a demo in some docs would work.  Does that sound good?

Reply:
Regarding splitting the expand button into a separate component - In case we agree on keeping this component, I would say it's a good idea. The button component doesn't have to be (and probably should not be) a separate section in our docs, it can be still just part of ExpandableText docs.

Also agree that without the internal state, the component probably does not make sense to exist. So I would vote for either keeping the component with its internal state (use it or implement your own) or dropping it completely. Maybe since some of our frontends are actively using it, I am a little more for keeping.

Reply:
If we pull the button out we are going to have to take the state with it.  Due to that I'm suggesting either getting rid of this component.  I think a demo would be better.

Reply:
This implementation seems very similar to the Expandable section "Truncate expansion" example found here https://www.patternfly.org/components/expandable-section#truncate-expansion

Reply:
If I'm not mistaken, we wouldn't have to move the state to the button, it could only be consumed as a boolean flag according to which it would display its corresponding variant. When clicked, it would then just call the function of the parent component, which would change the state.

Reply:
Okay I'll close this PR, and get a demo put together somewhere.